packages/esm-appointments-app/src/patient-queue/visit-form/visit-form.component.tsx

```diff
@@ -170,7 +171,7 @@ const VisitForm: React.FC<VisitFormProps> = ({ patientUuid, appointment }) => {
           },
         );
     },
-    [visitType, visitTime, timeFormat, patientUuid, visitDate, selectedLocation, t, mutate, currentAppointmentDate],
+    [visitType, visitTime, timeFormat, patientUuid, visitDate, selectedLocation, t, mutate, selectedDate],
```

Comment:
Minor: I'm not convinced we really need `useCallback` here (or in a lot of place). See [this](https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere). It makes it hard to maintain correctly (because of this dependency array)

Reply:
Good point and makes sense... I'm not going to touch this at this point because it's existing code that I might not be able to full test, but I'll be on the lookout for this in the future.